### ISSUE TITLE
A11Y: DataTableSkeleton component needs aria-live/ role.

### ISSUE THREAD

**Issue by sandhya-r-reddy** [carbon-design-system/carbon#2565]
Detailed description

JAWS screen reader doesn't inform the user when the DataTableSkeleton is used  to show the progressive loading state.

Is this issue related to a specific component?
DataTableSkeleton

What did you expect to happen? What happened instead? What would you like to see changed?
If the DataTableSkeleton is displayed to show the loading state,  screen reader should inform user about loading.

What browser are you working in?
Chrome, Firefox(Jaws)
What version of the Carbon Design System are you using?
6.39.3
What offering/product do you work on? Any pressing ship or release dates we should be aware of?
Genesis UI
